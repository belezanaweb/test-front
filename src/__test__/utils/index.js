import checkPropTypes from 'check-prop-types'

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`)
  return wrapper
}

export const checkProps = (component, expectedProps) => {
  const propsError = checkPropTypes(component.propTypes, expectedProps, 'props', component.name)
  return propsError
}
