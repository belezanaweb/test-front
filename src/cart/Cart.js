import React, { Component } from 'react';

class Cart extends Component {
  render() {
    return (
      <div>
        Cart
      </div>
    );
  }
}

export default Cart;