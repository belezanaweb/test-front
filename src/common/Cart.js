import axios from 'axios';

import DataStore from './DataStore';

const url = '//www.mocky.io/v2/5b15c4923100004a006f3c07';
const dataStoreCardId = 'CartObjProducts';

const Cart = {

  getCache: () => DataStore.get(dataStoreCardId),

  get: async (force) => {

    let objProducts;

    if (!force) objProducts = DataStore.get(dataStoreCardId);

    if (force || !objProducts) {

      try {

        const { data } = await axios.get(url);
        objProducts = data || {};

      } catch (e) {

        objProducts = {};

      }

      DataStore.set(dataStoreCardId, objProducts);

    }

    return objProducts;

  },

};

export default Cart;
