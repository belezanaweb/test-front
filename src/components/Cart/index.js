import React from 'react';
import {Container, ButtonPay} from "./styles"
import Total from"../Total"
import { Title, Section } from "../styles"
import CartList from "../CartList"

function Cart() {
  
  const messageButton='Seguir para o pagamento';
  
     
    return (

      <Section>
        <Title>Produtos</Title>
     <Container>             
         <CartList/>
     </Container>    
     <Total />  
     <ButtonPay type="button" href={"/payment"}>
          <p>{messageButton}</p>       
       </ButtonPay>       
      
     </Section>
    );
  }

export default Cart 