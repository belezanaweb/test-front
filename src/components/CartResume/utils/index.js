export const normalizeCartData = data => {
  return {
    productsQuantity: data.items.length,
    subTotal: data.subTotal.toFixed(2).replace('.', ','),
    shipping: data.shippingTotal.toFixed(2).replace('.', ','),
    discount: data.discount.toFixed(2).replace('.', ','),
    total: data.total.toFixed(2).replace('.', ','),
  };
};
