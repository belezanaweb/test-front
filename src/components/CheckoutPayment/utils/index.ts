import * as Yup from 'yup'

const errors = {
  name: 'Insira um nome válido',
  code: 'Insira um código válido',
  number: 'Insira um número válido',
  expiration: 'Insira uma data válida',
}

export const initialValues = {
  creditCardName: '',
  creditCardNumber: '',
  creditCardExpiration: '',
  creditCardCode: '',
}

const regexOnlyNumbers = /[^\d.-]+/g

export const CheckoutPaymentSchema = Yup.object().shape({
  creditCardName: Yup.string()
    .min(2, errors.name)
    .matches(/^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi, errors.name)
    .required(errors.name),
  creditCardNumber: Yup.string()
    .min(2, errors.number)
    .test('len', errors.number, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 16
    })
    .required(errors.number),
  creditCardExpiration: Yup.string()
    .min(2, errors.name)
    .test('len', errors.expiration, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 4
    })
    .required(errors.expiration),
  creditCardCode: Yup.string()
    .matches(/^[0-9]+$/, errors.code)
    .test('len', errors.code, (val) => val.length === 3)
    .required(errors.code),
})
