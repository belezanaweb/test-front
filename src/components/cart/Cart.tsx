import { Item } from "../../models/Item"
import { ListItem } from "./ListItem"
import { CartContainer } from "./styles"

type CartType = {
  items: Item[]
}

const Cart = ({ items }: CartType) => {
  return <CartContainer>
    {items.map((item) => (
      <ListItem
        key={item.product.sku}
        name={item.product.name}
        image={item.product.imageObjects[0].thumbnail}
        oldPrice={item.product.priceSpecification.maxPrice}
        price={item.product.priceSpecification.price}
      />
    ))}
  </CartContainer>
}

export { Cart }
