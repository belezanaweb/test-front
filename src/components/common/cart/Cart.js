import axios from 'axios';

import DataStore from '../dataStore/DataStore';

const url = '//www.mocky.io/v2/5b15c4923100004a006f3c07';
const dataStoreCardId = 'CartObjProducts';

const Cart = {

  shared: () => {

    const data = DataStore.shared(dataStoreCardId);

    return data;

  },

  readOnly: () => DataStore.readOnly(dataStoreCardId),

  download: async () => {

    // return {};

    let objProducts;

    try {

      /**
       * Warning: during the development the url was unavailable several times
       */
      const { data } = await axios.get(url);
      objProducts = data || {};

    } catch (error) {

      // TODO: do a better error handling
      return { error: 'Erro acessando API do carrinho' };

    }

    return DataStore.store(dataStoreCardId, objProducts);

  },

};

export default Cart;
