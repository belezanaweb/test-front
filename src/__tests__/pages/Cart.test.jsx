import React from 'react'
import { render } from '../../utils/test-utils'
import Cart from '../../pages/Cart'

jest.mock('../../hooks/checkout.jsx', () => ({
  useCheckout: jest.fn().mockReturnValue({
    cartItems: [],
    cartSummary: {},
    isLoading: false
  })
}))

describe('Cart Page', () => {
  test('should render without errors', () => {
    const { container } = render(<Cart />)
    expect(container).toBeInTheDocument()
  })

  test('should match with snapshot', () => {
    const { container } = render(<Cart />)
    expect(container).toMatchSnapshot()
  })
})
