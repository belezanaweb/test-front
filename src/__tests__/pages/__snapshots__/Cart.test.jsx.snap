// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Page should match with snapshot 1`] = `
<div>
  <main
    class="sc-eCApnc iHOios"
  >
    <header
      class="sc-bdnxRM cZjwSm"
    >
      <div
        class="sc-gtsrHT egWxvy"
      >
        <nav
          class="sc-dlnjwi kDoCbM"
        >
          <a
            class="sc-hKFxyN vGRFp"
            href="/cart"
          >
            Sacola
          </a>
          <a
            class="sc-hKFxyN vGRFp"
            href="/payment"
          >
            Pagamento
          </a>
          <a
            class="sc-hKFxyN vGRFp"
            href="/confirmation"
          >
            Confirmação
          </a>
        </nav>
      </div>
    </header>
    <div
      class="sc-jSFjdj drOsWB"
    >
      <section
        class="sc-iCoGMd"
      >
        <h2
          class="sc-fujyAs djPjrY"
        >
          Produtos
        </h2>
        <div
          class="sc-pNWdM fRwEYN"
        >
          <div
            class="sc-jrsJWt dqothG"
          >
            <ul />
          </div>
        </div>
      </section>
      <section
        class="sc-ksluID iNvcqR"
      >
        <div
          class="sc-fFSPTT sc-bkbkJK jPYEnV"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Produtos
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
        <div
          class="sc-fFSPTT sc-iemWCZ jPYEnV"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Frete
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
        <div
          class="sc-fFSPTT sc-dIvrsQ jPYEnV hRyHfU"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Desconto
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
        <div
          class="sc-fFSPTT sc-hHEiqL jPYEnV cFkeXV"
        >
          <p
            class="sc-hBMUJo kBEAzC"
          >
            Total
          </p>
          <p
            class="sc-fnVZcZ gLXOoI"
          >
            R$ NaN
          </p>
        </div>
      </section>
      <button
        class="sc-gKAaRy iuxzqw"
      >
        Seguir para o Pagamento
      </button>
    </div>
  </main>
</div>
`;
